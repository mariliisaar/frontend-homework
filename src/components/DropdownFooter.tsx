import { useDispatch } from 'react-redux'

const DropdownFooter = () => {
    const dispatch = useDispatch()

    return(
        <div className="nav__menu-footer">
            <div className="nav__menu-footer-links">
                <div className="nav__menu-link" onClick={() => dispatch({type: 'dropdown/close'})}>
                    <i className="material-icons-outlined nav__menu-link-icon">
                        settings
                    </i>
                    <div className="nav__menu-link-text">
                        Settings
                    </div>
                </div>
                <div className="nav__menu-link logout" onClick={() => dispatch({type: 'dropdown/close'})}>
                    <i className="material-icons-outlined nav__menu-link-icon">
                        exit_to_app
                    </i>
                    <div className="nav__menu-link-text">
                        Log out
                    </div>
                </div>
            </div>
        </div>
    )
}

export default DropdownFooter