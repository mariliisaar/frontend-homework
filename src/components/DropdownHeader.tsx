import { useAppSelector } from '../hooks'
import { useDispatch } from 'react-redux'

const DropdownHeader = () => {
    const active = useAppSelector( state => state.company.active)
    const dispatch = useDispatch()
        
    return (
        <div className="nav__menu-header">
            <div className="nav__menu-header-heading">
                Your companies
            </div>
            <div className={`nav__menu-link nav__menu-header-text ${active === 0 ? "nav__menu-active" : ""}`} onClick={() => dispatch({
                type: 'dropdown/company',
                payload: 0
            })}>
                Viljatootja AS
                <i className={`material-icons-outlined nav__menu-active-icon ${active === 0 ? "visible" : ""}`}>
                    check
                </i>
            </div>
            <div className={`nav__menu-link nav__menu-header-text ${active === 1 ? "nav__menu-active" : ""}`} onClick={() => dispatch({
                type: 'dropdown/company',
                payload: 1
            })}>
                Tartu Agro OÃœ
                <i className={`material-icons-outlined nav__menu-active-icon ${active === 1 ? "visible" : ""}`}>
                    check
                </i>
            </div>
        </div>
    )
}

export default DropdownHeader