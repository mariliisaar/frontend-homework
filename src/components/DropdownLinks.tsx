import { useDispatch } from 'react-redux'

const DropdownLinks = () => {
    const dispatch = useDispatch()

    return (
        <div className="nav__menu-links">
            <div className="nav__menu-link" onClick={() => dispatch({type: 'dropdown/close'})}>
                <i className="material-icons-outlined nav__menu-link-icon">
                    phone_iphone
                </i>
                <div className="nav__menu-link-text">
                    Get the mobile app
                </div>
            </div>
            <div className="nav__menu-link" onClick={() => dispatch({type: 'dropdown/close'})}>
                <i className="material-icons-outlined nav__menu-link-icon">
                    people
                </i>
                <div className="nav__menu-link-text">
                    Community
                </div>
            </div>
            <div className="nav__menu-link" onClick={() => dispatch({type: 'dropdown/close'})}>
                <i className="material-icons-outlined nav__menu-link-icon">
                    class
                </i>
                <div className="nav__menu-link-text">
                    Knowledge base
                </div>
            </div>
        </div>
    )
}

export default DropdownLinks