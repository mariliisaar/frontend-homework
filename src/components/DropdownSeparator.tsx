const DropdownSeparator = () => (
    <div className="nav__menu-separator"></div>
)

export default DropdownSeparator