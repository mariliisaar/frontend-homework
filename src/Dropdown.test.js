import { Provider } from 'react-redux'
import { shallow } from 'enzyme'
import configureMockStore from 'redux-mock-store'
import DropdownHeader from './components/DropdownHeader'
import DropdownLinks from './components/DropdownLinks'
import DropdownFooter from './components/DropdownFooter'
import DropdownSeparator from './components/DropdownSeparator'

describe('render dropdown footer', () => {
    const testStore = configureMockStore([])
    let store;

    beforeEach(() => {
        store = testStore({
            company: {
                name: 'Firma',
                active: 0
            },
            toggle: {
                open: false
            }
        })
    })

    it('renders header', () => {
      expect(shallow(<Provider store={store}><DropdownHeader /></Provider>)).toMatchSnapshot()
    })

    it('renders links', () => {
        expect(shallow(<Provider store={store}><DropdownLinks /></Provider>)).toMatchSnapshot()
    })

    it('renders footer', () => {
        expect(shallow(<Provider store={store}><DropdownFooter /></Provider>)).toMatchSnapshot()
    })

    it('renders separator', () => {
        expect(shallow(<Provider store={store}><DropdownSeparator /></Provider>)).toMatchSnapshot()
    })
})