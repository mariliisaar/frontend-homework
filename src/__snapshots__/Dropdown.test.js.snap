// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render dropdown footer renders footer 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <DropdownFooter />
</ContextProvider>
`;

exports[`render dropdown footer renders header 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <DropdownHeader />
</ContextProvider>
`;

exports[`render dropdown footer renders links 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <DropdownLinks />
</ContextProvider>
`;

exports[`render dropdown footer renders separator 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <DropdownSeparator />
</ContextProvider>
`;
